### ISSUE TITLE
Support fallback release mechanism if commits aren't formalized

### ISSUE THREAD

**Issue by james-r-smith** [semantic-release/semantic-release#1294]
## New feature motivation

I'm coming from a place where the concept of CI/CD is not widely practiced (this may apply to other developers as well 🙌). In addition, having to force developers to change the way that they commit so that this tool will work correctly is not a reasonable option to bring to the table.

My request is to support this initial phase of onboarding of semantic releases by introducing a fallback patch release for any commits to master. Then, as incentive for starting to have a better changelog, the ability to automagically release major / minor versions, and all the other additional benefits, the developers would have to practice "better" (objective opinion 👌) commit messages.

Think of it as a "Want to introduce your team to automatic semantic releases? Here's how to start..." 😊 

## New feature description

The new feature would construct releases for **any** and **all** commits to master. This means that even if the commit messages have failed the formal structure, a **fallback patch release** will still be done (while not being able to reap the benefits that formal commit messages bring).

**TL;DR:** If a commit is done on master and doesn't follow the formal message structure, a fallback patch release will be done.

## New feature implementation

Changes will probably need to be done somewhere around **[here](https://github.com/semantic-release/semantic-release/blob/v15.13.24/index.js#L91-L94)**.

**Comment by gr2m** [comment]
This should be possible with a plugin as a replacement for https://github.com/semantic-release/commit-analyzer

**Comment by james-r-smith** [comment]
@gr2m, I did some digging last night, and I have found a solution for a patch release fallback:

```js
{header: /.*/, release: 'patch'}
```

This will satisfy the needs requested. 😊

**Comment by pvdlg** [comment]
That's indeed what I would recommend. Every commit will trigger a release this way.

**Comment by adamyonk** [comment]
I was wanting something similar, but as a *fallback*. This config got me there:
```
const defaultRules = require('@semantic-release/commit-analyzer/lib/default-release-rules.js');

module.exports = {
  plugins: [
    [
      '@semantic-release/commit-analyzer',
      {
        preset: 'angular',
        releaseRules: [
          ...defaultRules,
          // Use the default rules, and fall back to patching everything
          { release: 'patch' },
        ],
      },
    ],
    '@semantic-release/release-notes-generator',
    '@semantic-release/npm',
    '@semantic-release/github',
  ],
};
```
